### PR TITLE
Fix QGCSwitch  and ClickableColor after migration to Qt6

### PR DESCRIPTION
**Pull request by s-lisovenko**
After migration to Qt6 these controls were broken

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
Doesn't Switch need to be changed to backgroundItem and contentItem: https://doc.qt.io/Qt-6/qtquickcontrols-customize.html#customizing-switch

**Comment by s-lisovenko** [comment]
> Doesn't Switch need to be changed to backgroundItem and contentItem: https://doc.qt.io/Qt-6/qtquickcontrols-customize.html#customizing-switch

Do you mean using SwitchDelegate instead switch? This is what i see right now with switch.
https://github.com/mavlink/qgroundcontrol/assets/2522054/7464fc98-5bb9-4df3-8602-cce2a0df2574

**Comment by s-lisovenko** [comment]
Delete duplicate imports for QGCSwitch

**Comment by DonLakeFlyer** [comment]
> Do you mean using SwitchDelegate instead switch?

No. In your change it look like you have the the first Rectangle which is actually the background for the control. This should be in the backgroundItem property. And then the indicator is a child of that first Rectangle. This part should be the contentItem. See the link I included on how to style the Switch control in Qt6.

**Comment by s-lisovenko** [comment]
Add background item to the switch control

**Comment by s-lisovenko** [comment]
> > Do you mean using SwitchDelegate instead switch?
> 
> No. In your change it look like you have the the first Rectangle which is actually the background for the control. This should be in the backgroundItem property. And then the indicator is a child of that first Rectangle. This part should be the contentItem. See the link I included on how to style the Switch control in Qt6.

Added background item. I don't think the content item is needed here. So we have separate background and indicator. Please check again

**Comment by DonLakeFlyer** [comment]
> I don't think the content item is needed here.

To be completely correct it does. The previous implementation didn't have it which was wrong. Without the the Text will not the themed correctly to the QGC palette.

**Comment by s-lisovenko** [comment]
> To be completely correct it does. The previous implementation didn't have it which was wrong. Without it the Text will not the themed correctly to the QGC palette.

Sorry I think I misunderstood you. Do yo want to display text information next to a switch? now it's just a stylized switch as in  before QGC implementation? Perhaps you can fix this according to your or the core team's vision? But I think that having the option of a simple switch without text should be left as well.

**Comment by DonLakeFlyer** [comment]
> now it's just a stylized switch as in before QGC implementation

The previous version supported text as well (incorrectly though), since it used the regular Qml Switch as the base control. If you didn't want text you just didn't specify it.

**Comment by s-lisovenko** [comment]
Screenshot from 2024-01-27 21-42-45

Added the ability to display text to QGCSwitch
@DonLakeFlyer please take a look.

File: src/QmlControls/QGCSwitch.qml [hunk 1]
```diff
@@ -9,20 +9,43 @@
...
+    id: control
+
+    readonly property int _radius: 3
+
+    QGCPalette {
+        id:                 qgcPal
+        colorGroupEnabled:  true
+    }
+
+    contentItem: Text {
```
Comment (DonLakeFlyer) [inline src/QmlControls/QGCSwitch.qml:26]: This should just be QGCLabel. That way it will automatically follow any styling changes.

Comment (s-lisovenko) [reply src/QmlControls/QGCSwitch.qml]: Yes, changed

**Comment by DonLakeFlyer** [comment]
Thanks for the fix

**Pull request merged**